test/extended/syncer/syncer_utils.go

```diff
@@ -0,0 +1,136 @@
+func (s *SyncTarget) CreateAsExpectedResult(k *exutil.CLI, successFlag bool, containsMsg string) {
+	kcpServerVersion, getError := exutil.GetKcpServerVersion(k)
+	o.Expect(getError).ShouldNot(o.HaveOccurred())
+	o.Expect(kcpServerVersion).Should(o.MatchRegexp(`v\d+(.\d+){0,2}`))
+	msg, createError := k.WithoutNamespace().WithoutKubeconf().WithoutWorkSpaceServer().Run("kcp").Args("--server="+s.WorkSpaceServer, "workload", "sync", s.Name,
+		"--syncer-image=ghcr.io/kcp-dev/kcp/syncer:"+kcpServerVersion, "--output-file="+s.OutputFilePath).Output()
+	if successFlag {
+		o.Expect(createError).ShouldNot(o.HaveOccurred())
+		o.Expect(msg).Should(o.ContainSubstring(containsMsg))
+	} else {
+		o.Expect(createError).Should(o.HaveOccurred())
+		o.Expect(fmt.Sprint(msg)).Should(o.ContainSubstring(containsMsg))
+	}
+}
+
+// GetFieldByJSONPath gets specific field value of the SyncTarget by jsonpath
+func (s *SyncTarget) GetFieldByJSONPath(k *exutil.CLI, JSONPath string) (string, error) {
+	return k.WithoutNamespace().WithoutKubeconf().WithoutWorkSpaceServer().Run("get").Args("--server="+s.WorkSpaceServer, "syncTarget", s.Name, "-o", "jsonpath="+JSONPath).Output()
+}
+
+// CheckReady checks whether the SyncTarget is ready
+func (s *SyncTarget) CheckReady(k *exutil.CLI) (bool, error) {
+	readyFlag, err := s.GetFieldByJSONPath(k, `{.status.conditions[?(@.type=="Ready")].status}`)
+	if err != nil {
+		e2e.Logf(`Get SyncTarget/%s status faied: "%v"`, s.Name, err)
+		return false, err
+	}
+	e2e.Logf("SyncTarget/%s ready condition is %s", s.Name, readyFlag)
+	return strings.EqualFold(readyFlag, "True"), nil
+}
+
+// WaitUntilReady waits the SyncTarget become ready
+func (s *SyncTarget) WaitUntilReady(k *exutil.CLI) {
+	err := wait.Poll(5*time.Second, 180*time.Second, func() (bool, error) {
+		SyncTargetReady, err := s.CheckReady(k)
+		if err != nil {
+			return SyncTargetReady, err
+		}
+		return SyncTargetReady, nil
+	})
+	exutil.AssertWaitPollNoErr(err, fmt.Sprintf("Waiting SyncTarget/%s become ready timeout", s.Name))
+}
+
+// CheckDisplayAttributes checks the SyncTarget info showing the expected columns
+func (s *SyncTarget) CheckDisplayAttributes(k *exutil.CLI) {
+	syncTargetInfo, err := k.WithoutNamespace().WithoutKubeconf().WithoutWorkSpaceServer().Run("get").Args("--server="+s.WorkSpaceServer, "synctarget", s.Name).Output()
+	o.Expect(err).NotTo(o.HaveOccurred())
+	o.Expect(syncTargetInfo).Should(o.And(
+		o.ContainSubstring("NAME"),
+		o.ContainSubstring("AGE"),
+	))
+	syncTargetInfo, err = k.WithoutNamespace().WithoutKubeconf().Run("get").Args("synctarget", s.Name, "-o", "wide").Output()
+	o.Expect(err).NotTo(o.HaveOccurred())
+	o.Expect(syncTargetInfo).Should(o.And(
+		o.ContainSubstring("NAME"),
+		o.ContainSubstring("LOCATION"),
+		o.ContainSubstring("READY"),
+		o.ContainSubstring("SYNCED"),
+		o.ContainSubstring("API"),
+		o.ContainSubstring("RESOURCES"),
+		o.ContainSubstring("KEY"),
+		o.ContainSubstring("AGE"),
+	))
+}
```

Comment:
@Phaow "SYNCED, API, RESOURCES " are not three columns and they are just one. Can we make sure that columns does contain values ?There was a bug where SYNCED API RESOURCES was not indicating anything.

[knarra@knarra ~]$ kubectl get synctarget -o wide
NAME LOCATION READY SYNCED API RESOURCES KEY AGE
qecluster1 qecluster1 False cfKajMFOBq2wSWpukT9xyoCBcSqs1NRRFAaytK 4d3h

Reply:
Got it. Thanks for clarify! I'll enhance it.

Reply:
Here is the issue https://github.com/kcp-dev/kcp/issues/943

Reply:
Thanks for the info!

Reply:
Addressed, but as it has a known issue, I'll comment the checkpoint firstly.